diff --git a/lib/galaxy/jobs/handler.py b/lib/galaxy/jobs/handler.py
index acaaa1c10e..a203790f68 100644
--- a/lib/galaxy/jobs/handler.py
+++ b/lib/galaxy/jobs/handler.py
@@ -572,6 +572,18 @@ class JobHandlerQueue(Monitors):
                 log.debug("Intentionally failing job with message (%s)" % failure_message)
             job_wrapper.fail(failure_message)
             return JOB_ERROR, job_destination
+
+        try:
+            failure_message= 'You do not seem to be logged in, and can therefore not submit jobs.'
+            assert job.user is not None, failure_message
+        except AssertionError as e:
+            failure_message= 'You do not seem to be logged in, and can therefore not submit jobs.'
+            log.debug("Intentionally failing job with message (%s)" % failure_message)
+            job_wrapper.fail(failure_message)
+            state = JOB_ERROR
+            return JOB_ERROR, job_destination
+            
+            
