diff --git a/lib/galaxy/jobs/handler.py b/lib/galaxy/jobs/handler.py
index f5b98d61f4..bc7aace9bf 100644
--- a/lib/galaxy/jobs/handler.py
+++ b/lib/galaxy/jobs/handler.py
@@ -604,7 +604,11 @@ class JobHandlerQueue(Monitors):
                     elif history.state == "ok":
                         finished = history.create_time

-                time_spent += finished - started
+                if started is not None and finished is not None:
+                    time_spent += finished - started
+                else:
+                    log.warning(
+                        f"Unable to calculate time spent for job {job.id}; started: {started}, finished: {finished}")

             if time_spent > self.app.job_config.limits.total_walltime["delta"]:
                 return JOB_USER_OVER_TOTAL_WALLTIME, job_destination
